### PR TITLE
Added Zippy Courses integration for AffiliateWP

### PR DESCRIPTION
**Pull request by jwondrusch**
I spoke with @pippinsplugins 2 days ago about including ZippyCourses support in the next release of AWP (he mentioned at the end of this month).  Here is a PR with a jump start on the integration (it's working on our test servers with our various gateways).

Created and tested AWP + ZippyCourses integration.  Please contact support@zippycourses.com if you need a copy of Zippy Courses to test with, or if we can do anything else to expedite the inclusion of Zippy Courses as an integration with AWP.

### CODE REVIEW

File: includes/integrations/class-zippycourses.php [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        add_action( 'save_post', array( $this, 'save_meta' ) );
+
+        add_action( 'zippy_event_order_status_change', array( $this, 'add_pending_referral' ), 10 );
+        add_action( 'zippy_event_order_status_change', array( $this, 'mark_referral_complete' ), 10 );
+        add_action( 'zippy_event_order_status_change', array( $this, 'revoke_referral_on_refund' ), 10 );
+
+        add_filter( 'affwp_referral_reference_column', array( $this, 'reference_link' ), 10, 2 );
+    }
+
+    public function metabox()
```
Comment (pippinsplugins) [inline includes/integrations/class-zippycourses.php]: Nit picky but please place the  bracket on the same line as the function definition.

File: includes/integrations/class-zippycourses.php [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        }
+
+        $post = get_post( $post_id );
+
+        if( ! $post ) {
+            return $post_id;
+        }
+
+        // Check post type is product
+        if ( 'product' != $post->post_type ) {
```
Comment (pippinsplugins) [inline includes/integrations/class-zippycourses.php]: Is `product` the proper post type here?

Comment (jwondrusch) [inline includes/integrations/class-zippycourses.php]: Yep, it is :)

**Comment by jwondrusch** [comment]
Made the metabox style update :) Sorry about that, we use PSR-2 for Zippy Courses, so slipped up in trying to match the EDD style.

**Comment by pippinsplugins** [comment]
Excellent, looks good.

I'd like to get a copy of the plugin. Sending an email now.

**Comment by pippinsplugins** [comment]
@sumobi plugin is in dropxbox.

**Comment by sumobi** [comment]
@pippinsplugins :+1:

**Comment by jwondrusch** [comment]
Are we looking good for inclusion in the next AWP release?

**Comment by pippinsplugins** [comment]
@jwondrusch Yes, that's the goal.

**Comment by sumobi** [comment]
@jwondrusch @pippinsplugins 

Looking good so far! Couple of things I spotted:

1. The referral doesn't seem to be rejected when I change the order status to `Refunded`. `Reject Unpaid Referrals on Refund?` is enabled. Can anyone else replicate?

    screen shot 2015-09-02 at 2 13 03 pm
___
2. Would be nice to show the affiliate and how much they were awarded on the order screen, similar to other integrations. eg:

    screen shot 2015-09-02 at 2 16 11 pm

**Comment by jwondrusch** [comment]
I should be able to tackle both of those in between support tickets tomorrow.  What's the deadline for finishing those for inclusion in the next release?

**Comment by sumobi** [comment]
@jwondrusch Awesome thanks. We don't have a deadline (or exact release date) so you'll be fine. Just sometime within the next week if you can.

**Comment by jwondrusch** [comment]
Just sent @pippinsplugins an updated version of the Zippy Courses plugin that includes the the fixes requested.

Some of our customers have been excitedly awaiting the release / integration with AffiliateWP :) Is there a ballpark I can give them for the next release or more of a, "When it's done"?

**Comment by pippinsplugins** [comment]
@sumobi updated copy in Dropbox

**Comment by sumobi** [comment]
Awesome, will do some more testing.

@jwondrusch https://twitter.com/affwp/status/639602461703974912

**Comment by jwondrusch** [comment]
Awesome, thank you @sumobi :) Was fun to see.

**Comment by sumobi** [comment]
@jwondrusch :)

Just did some more testing, it seems referrals are being generated with a `rejected` status, but only if `Reject Unpaid Referrals on Refund?` is enabled.

Steps to reproduce:

1. Make sure `Reject Unpaid Referrals on Refund?` is enabled in Affiliates -> Settings -> Misc

2. Use an affiliate's link to store the visit in Affiliates -> Visits

3. Purchase a product using Stripe gateway

4. The order in Zippy Courses is created and the status is shown as `completed`. The referral however is shown as `rejected` and the visit from Affiliates -> Visits is not shown as converted.

Nice job on the notes :+1: thanks for including that.
screen shot 2015-09-07 at 6 25 35 pm

If you disable `Reject Unpaid Referrals on Refund?` then everything seems to work as expected. The visit is shown as converted as soon as a payment is made and the referral is created with a status of `unpaid`.

One more thing worth mentioning. There's a bit of a UI glitch, can you replicate the below?

If you make your browser window smaller, things start to overlap:

screen shot 2015-09-07 at 6 01 18 pm
___

If you make the browser window rather wide, things come right:

screen shot 2015-09-07 at 6 01 40 pm

**Comment by jwondrusch** [comment]
Thanks, working on this now, and will have a new PR today.

**Comment by sumobi** [comment]
@jwondrusch all looks good to me! How could I test a "pending" order in Zippy Courses so I can test a pending referral being created? I tried paying with PayPal but no pending order was created when I reached the PayPal site.

@pippinsplugins want to have a play before merging?

**Comment by jwondrusch** [comment]
@sumobi That's awesome news :)

"Pending" is a hard "window" to hit - Pending orders are mostly for "waiting" periods - an example being when Zippy Courses has received a notification of the subscription creation but not the payment from PayPal via IPN.

We use Transactions to store more specific data, which also have statuses, but we've chosen to tie the integration into order status, as that's how our permissions are controlled, and is how we would like to associate referrals with Zippy.

**Comment by pippinsplugins** [comment]
Merged!

**Comment by jwondrusch** [comment]
Woohoo, thanks guys!

**Pull request merged**